### DISCUSSION TITLE
"fixup -C" during interactive rebase within magit?

### DISCUSSION THREAD

**Discussion by mdempsky** [magit/magit#4747] [Ideas]
I often want to squash two commits with fixup, but keep the *second* commit's message rather than the first. Just now I noticed that git evidently supports this with the "fixup -C" command:

```
# f, fixup [-C | -c] <commit> = like "squash" but keep only the previous
#                    commit's log message, unless -C is used, in which case
#                    keep only this commit's message; -c is same as -C but
#                    opens the editor
```

However, it doesn't seem like magit has any builtin way for me to actually use this feature.

I hacked it up myself with:

```
(defun git-rebase-fixup-C ()
  "Meld commit on previous line into current commit, discard its message.
If the region is active, act on all lines touched by the region."
  (interactive)
  (git-rebase-set-action "fixup -C"))

(define-key git-rebase-mode-map (kbd "F") #'git-rebase-fixup-C)
```

This works okay, but isn't perfect. E.g., if I press `F` to change a line to `fixup -C 12345...` and then press `c`, it changes to `pick -C 12345...` instead of just `pick 12345...`.

**Comment by kyleam** [comment]
> Just now I noticed that git evidently supports this with the "fixup -C" command:

Yes, that was added relatively recently in v2.32.0.

Adjusting git-rebase.el to support it has been on my to-do list, but offhand I don't know how involved that will be.

**Reply by tarsius** [reply]
This does indeed sound useful.

**Reply by tarsius** [reply]
I've added support for this in v4.3.6 (9674c4755a9871568c780beb53bde9ad472db067).
